src/components/shared/ListView/ListView.tsx

```diff
@@ -0,0 +1,212 @@
+    shadow="xl"
+  >
+    <Heading as="h3" fontSize="lg">
+      {o.Name.length > 39 ? o.Name.substring(0, 39) + "..." : o.Name}
+    </Heading>
+  </VStack>
+)
+
+const DEFAULT_GRID_OPTIONS = {templateColumns: "repeat(3, 1fr)", templateRows: "(3, 1fr)", gap: 4}
+
+const ListView = <T,>({
+  service,
+  columns,
+  paramMap,
+  queryMap,
+  gridOptions = DEFAULT_GRID_OPTIONS,
+  renderCard = DEFAULT_CARD,
+  children
+}: ListViewProps<T>) => {
+  const [data, setData] = useState<(T extends Product ? ListPageWithFacets<T> : ListPage<T>) | undefined>()
+  const [viewMode, setViewMode] = useState<"grid" | "table">("table")
+  const [selected, setSelected] = useState<string[]>([])
+
+  const handleSelectChange = useCallback((id: string, isSelected: boolean) => {
+    setSelected((s) => (isSelected ? [...s, id] : s.filter((sid) => sid !== id)))
+  }, [])
+
+  const {push, pathname, isReady, query} = useRouter()
+
+  const mapRouterQuery = useCallback(
+    (map?: {[key: string]: string}) => {
+      let result = {}
+      if (!isReady) return result
+      if (!map) return result
+      Object.entries(query).forEach(([key, val]: [string, string]) => {
+        if (map[key]) {
+          result[map[key]] = val
+        }
+      })
+      return result
+    },
+    [query, isReady]
+  )
+
+  const params = useMemo(() => {
+    return {
+      routeParams: mapRouterQuery(paramMap),
+      queryParams: mapRouterQuery(queryMap)
+    }
+  }, [paramMap, queryMap, mapRouterQuery])
+
+  const fetchData = useCallback(async () => {
+    let response
+    if (Object.values(params.routeParams).length) {
+      response = await service(...Object.values(params.routeParams), params.queryParams)
+    } else {
+      response = await service(params.queryParams)
+    }
+    setData(response)
+  }, [service, params])
+
+  useEffect(() => {
+    if (isReady) {
+      fetchData()
+    }
+  }, [fetchData, isReady])
+
+  const viewModeToggle = useMemo(() => {
+    return (
+      <ButtonGroup isAttached variant="outline">
+        <IconButton
+          aria-label="Grid View"
+          isActive={viewMode === "grid"}
+          icon={<HiOutlineViewGrid />}
+          onClick={() => setViewMode("grid")}
+        />
+        <IconButton
+          aria-label="List View"
+          isActive={viewMode === "table"}
+          icon={<HiOutlineViewList />}
+          onClick={() => setViewMode("table")}
+        />
+      </ButtonGroup>
+    )
+  }, [viewMode])
+
+  const handleUpdateQuery = useCallback(
+    (queryKey: string) => (value: string | boolean | number) => {
+      push({pathname: pathname, query: {...query, [queryKey]: value}})
+    },
+    [push, pathname, query]
+  )
+
+  const metaInformationDisplay = useMemo(() => {
+    if (!data) return
+    return (
+      <Text
+        alignSelf="center"
+        flexShrink={0}
+        fontWeight="bold"
+      >{`${data.Meta.ItemRange[0]} - ${data.Meta.ItemRange[1]} of ${data.Meta.TotalCount}`}</Text>
+    )
+  }, [data])
```

Comment:
Instead of returning this component can we return metaInformation (raw data) so that implementers can craft their own display however they choose? We could still create a `MetaInformationDisplay` component that people could optionally use as a default. Realistically, we'd use this everywhere but it would let someone do per instance styling.

Reply:
Agreed. I'll make this change.